HIP/hip-796.md

```diff
@@ -0,0 +1,728 @@
+- **Mirror-6**: As a mirror node user, I want to know all NFTs that are part of a partition of an account.
+
+### SDKs
+
+- **SDK-1**: As an SDK user, I want to have an API to tell me the lock, pause, and freeze status of any token (including partitions), and the number of tokens that are available for transfer.
+- **SDK-2**: As an SDK user, I want to have an API for working with partitions and locks as defined by the HAPI.
+
+### Wallets
+
+- **WALLET-1**: As a wallet user, I want my wallet to show me whether a given token (including partitions) are frozen or paused, and the number of tokens that are available for transfer.
+- **WALLET-2**: As a wallet user, I want my wallet to show me the relationship between partitions in my account and the `token-definition` that they belong to, and the metadata associated with each partition.
+
+### Mirror Node Explorer
+
+- **EXPLORER-1**: As a user of the mirror node explorer, I want to see, for any given account's tokens, the relationship of partitions held by that account, and the status of each partition or token (locked, unlocked, paused, frozen, etc).
+
+## Example
+
+The concept of partitioning and locking can be illustrated using an example. Consider a `token-definition` for `AcmeToken` that represents investments in a fund.
+
+- The token issuer creates this `token-definition` with a `partition-key`. Let’s say this has a `TokenID` of `0.0.123456`.
+- The fund has different tranches with three different timelines until which these funds are locked (i.e. cannot be transferred or traded by a holder). Let’s call these tranches Tranche-A (locked until Jan 1, 2024), Tranche-B (locked until Jan 1, 2025), and Tranche-C (locked until Jan 1, 2026).
+- The token issuer creates three `partition-definition`s for the token-definition `0.0.123456`. Hedera creates these three definitions and returns `TokenID`s of `0.0.200001`, `0.0.200002`, and `0.0.200003` that represent AcmeToken.Tranche-A, AcmeToken.Tranche-B, AcmeToken.Tranche-C respectively.
+- The token issuer sets three different metadata on these partition definitions independently of each other.
+- When anybody queries the mirror node for this token id `0.0.123456`, they get an equivalent of:
+
+```
+{
+    id: 0.0.123456,
+    name: AcmeToken,
+    symbol: ACME,
+    lockKey: 0x111abc..,
+    partitionKey: 0x222abc..,
+    partitionMoveKey: 0x333ace…,
+    partitions: [
+        {partition_number:1, metadata: “ipfs://xxx”, partition_id: 0.0.200001},
```

Comment:
What is the purpose of `partition_number`?  `partition_id` is a sufficient primary key.  If `partition_number` is required how is it assigned (appears nowhere else in this document)?